lineapy/graph_reader/session_artifacts.py

```diff
@@ -603,11 +603,11 @@ def _update_nodecollection_dependencies(self):
                 if art.name in nc_graph.nodes
             ]
 
-    def _get_first_artifact_name(self) -> Optional[str]:
+    def _get_first_artifact_name(self) -> str:
         """
         Return the name of first artifact(topologically sorted)
         """
         for coll in self.artifact_nodecollections:
             if coll.collection_type == NodeCollectionType.ARTIFACT:
                 return coll.safename
-        return None
+        return ""
```

Comment:
What's the rationale to change this to an empty string other than typing? I personally think if there is no artifact in the collection, it should be a None, not an empty string. Furthermore, we are actually allowing an empty string as an artifact name. 

For example
```
a = ''
lineapy.save(a, '')
```
is actually running.

Reply:
Reverting to returning `None` & fixing the relevant `mypy` issue differently.